blade-graphics/src/vulkan/resource.rs

```diff
@@ -265,13 +265,17 @@ impl crate::traits::ResourceDevice for super::Context {
             extent: super::map_extent_3d(&desc.size),
             mip_levels: desc.mip_level_count,
             array_layers: desc.array_layer_count,
-            samples: vk::SampleCountFlags::from_raw(1), // desc.sample_count
+            samples: vk::SampleCountFlags::from_raw(desc.sample_count),
             tiling: vk::ImageTiling::OPTIMAL,
             usage: map_texture_usage(desc.usage, desc.format.aspects()),
             sharing_mode: vk::SharingMode::EXCLUSIVE,
             ..Default::default()
         };
 
+        /*
+            TODO(ErikWDev): Support lazily allocated texture with transient allocation for efficient msaa?
```

Comment:
I don't think this needs to be brought up in the context of MSAA specifically. Supporting transient allocations would be useful in general.

Reply:
Yup no not intrinsically related.

It is extra important for bandwidth saving in MSAA on tiled-renderers which is why it popped up in my mind while doing this.

I want to measure though to see what the drivers do with general usage and allocations, though the vulkan samples seems to show with measurements that it makes a bandwidth difference:

https://docs.vulkan.org/samples/latest/samples/performance/msaa/README.html